controllers/datadogagent/component/clusteragent/default.go

```diff
@@ -128,6 +128,14 @@ func defaultEnvVars(dda metav1.Object) []corev1.EnvVar {
 				},
 			},
 		},
+		{
+			Name: apicommon.DDClusterAgentServiceAccountName,
```

Comment:
do you mind adding this here instead? https://github.com/DataDog/datadog-operator/blob/0b47ddc48b78712eaa4d642ab53ffa788ce9dbc1/controllers/datadogagent/feature/enabledefault/feature.go#L384

can use the serviceaccount name set [here](https://github.com/DataDog/datadog-operator/blob/0b47ddc48b78712eaa4d642ab53ffa788ce9dbc1/controllers/datadogagent/feature/enabledefault/feature.go#L118)

Reply:
I moved it, but unfortunately it made me remove the test on my env variable in `/controllers/datadogagent/clusteragent_test.go` as well since `ManageClusterAgent()` is not called by these tests (afaik). Should I keep it where it originally was to keep the test or is there many tests I didn't see ?

Reply:
many of our features are lacking tests, I saw you ended up adding it in enabledefault/feature_test.go. thank you for that!